dtbase/ingress/ingress_weather.py

```diff
@@ -0,0 +1,229 @@
+"""
+Python module to import data using the Openweathermap API
+"""
+import os
+import json
+import logging
+import requests
+from datetime import datetime, timedelta
+import pandas as pd
+
+from dtbase.core.constants import (
+    CONST_BACKEND_URL,
+    CONST_LAT,
+    CONST_LON,
+    CONST_OPENWEATHERMAP_APIKEY,
+    CONST_OPENWEATHERMAP_HISTORICAL_URL,
+    CONST_OPENWEATHERMAP_FORECAST_URL,
+)
+from dtbase.ingress.ingress_utils import (
+    backend_call,
+    log_rest_response,
+    add_sensor_types,
+    add_sensors,
+)
+
+# Mapping of Openweathermap metrics to sensor measures in the database.
+METRICS_TO_MEASURES = {
+    "temperature": {
+        "name": "temperature",
+        "units": "degrees Celsius",
+    },
+    "relative_humidity": {
+        "name": "relative humidity",
+        "units": "percent",
+    },
+    "air_pressure": {
+        "name": "air pressure",
+        "units": "millibar",
+    },
+    "wind_speed": {
+        "name": "wind speed",
+        "units": "m/s",
+    },
+    "wind_direction": {
+        "name": "wind direction",
+        "units": "degrees",
+    },
+    "rain": {
+        "name": "rain",
+        "units": "mms",
+    },
+    "icon": {
+        "name": "icon",
+        "units": "",
+    },
```

Comment:
This seems to create an extra plot which is empty and has an "icon" y-axis. Can that be deleted or is it a placeholder for something?

Reply:
So, "icon" is a string that tells applications what weather icon png to show (we used it in the Unity app for CROP).   I guess chart.js doesn't know how to plot strings.   We could add some logic to exclude string measures from the data that gets passed to the template, but I'm not sure it'd be worth the effort for the moment (though may revisit this...)